File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int tall;
    do
    {
        //take the height from the user

        tall = get_int("height:  ");
    }
    // don't proceed until the user enters a valid number

    while (tall <= 0 || tall > 8);

    // the outer loop is for the rows
    //the inner loop is for the columns

    for (int i = 1; i <= tall; i++)
    {
        for (int j = 1; j <= tall; j++)
        {
            if (j > tall - i)
            {
                printf("#");
                continue;
            }
            printf(" ");
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
# take the height from the user
# reject this input as invalid, as by re-prompting the user to type in another number.
flag = True
while flag:
    try:
        tall = int(input("height:"))
        if not tall <= 0 or tall > 8:
            flag = False
    except ValueError:
        continue

while True:
    if tall <= 0 or tall > 8:
        tall = int(input("height:"))
        continue
    else:
        break
# the outer loop is for the rows
# the inner loop is for the columns
for i in range(1, tall + 1):
    for j in range(1, 11 + i):
        if (j > tall - i) and (j <= tall):
            print("#", end="")
            continue
        if (j > tall + 2) and (j <= tall + 2 + i):
            print("#", end="")
            continue
        if j <= tall + 2:
            print(" ", end="")
    print()